apps/factory/factories.py

```diff
@@ -0,0 +1,203 @@
+import factory
+import factory.fuzzy
+import random
+import pytz
+import uuid
+
+
+from apps.profiles.models import ChaUser, Instructor, StudentMembership, AssistantMembership, PasswordResetRequest, GithubUserInfo
+from apps.homework.models import Definition, Submission, SubmissionTracker, Grade, Question, QuestionAnswer, Criteria, CriteriaAnswer, TeamCustomChallengeURL
+from apps.klasses.models import Klass
+from apps.groups.models import Group, Team
+
+
+class UserFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = ChaUser
+    username = factory.Faker('user_name')
+    email = factory.LazyAttribute(lambda o: f"{o.username}@example.com")
```

Comment:
Is this lazily referencing the previous factory output (IE: username from above) or is this making a new call? If so could just move the call up into a temp variable and do assignment that way.

Reply:
This is a lazy reference to the instance's username, so the email and the username are similar.